torchvision/prototype/features/_label.py

```diff
@@ -77,3 +77,21 @@ def new_like(
         return super().new_like(
             other, data, categories=categories if categories is not None else other.categories, **kwargs
         )
+
+    @classmethod
+    def from_category(
+        cls,
+        category: str,
+        *,
+        categories: Sequence[str],
+        **kwargs: Any,
+    ) -> Label:
+        ohe_tensor = torch.zeros(len(categories), dtype=torch.long)
+        ohe_tensor[categories.index(category)] = 1
+        return cls(ohe_tensor, categories=categories, **kwargs)
+
+    def to_categories(self) -> Any:
+        if self.categories is None:
+            raise RuntimeError("OneHotLabel does not have categories")
+        label = self.argmax(dim=-1)
```

Comment:
Is this an assumption we can safely make? IMO, that is very case specific. Imagine we have 60% dog and 40% cat. Do we really want to say this is 100% a dog for all cases? I left this method out when implementing `OneHotLabel` due to this issue.

Reply:
Ground truth is rarely presented as percentages of one class and another. If we have multilabel task (vs multiclass), think of music tags for songs for example, then I'd create a MultiLabel class for that. OneHotLabel is one-hot encoded label which is produced as int (class index) -> vector (0, ..., 0, 1, 0, ... 0), thus we can safely pass back from vector to int. Am I missing something ?

Reply:
So, technically it is not a one-hot-encoded label :)

Reply:
Good points.

Philip is right that this won't work very well for multi-class or soft-labels. Victor is also right to say "those 2 cases are not one-hot". 

But isn't our intention to use the specific class to store the info both for multi and soft-label cases? Isn't this what we do with MixUp and CutMix already?

So here I see two problems:
1. Probably Philip is right, those methods make assumptions that we might not make.
2. Victor is right, the name on this class is misleading.

Reply:
I agree, the naming is not optimal. Any ideas on how to improve? Do these concepts, i.e. representing the label through either an index or a percentage, have names in the literature?

Reply:
K-hot encoded for multi-label is used sometimes. Soft-assignments/labels for things that add to 100%. Off the top of my head not great names. I'm open to suggestions